ntp-daemon/src/peer.rs

```diff
@@ -55,9 +69,12 @@ pub async fn start_peer<A: ToSocketAddrs, C: 'static + NtpClock + Send>(
                             // TODO log something
                         } else {
                             let packet = NtpHeader::deserialize(&buf);
-                            let result = peer.handle_incoming(packet, timestamp);
 
-                            if peer.accept_synchronization(timestamp, NtpDuration::ZERO).is_err() {
+                            let system_snapshot = *system_snapshots.borrow_and_update();
```

Comment:
right now I'm waiting with reading the `system_snapshots` till the last moment. Does that actually matter. If not we could refactor a little bit